functional-tests/fn_bridge_withdraw_happy.py

```diff
@@ -0,0 +1,179 @@
+        lower_bound = ((rollup_deposit_amount - 1_000_000) * SATS_TO_WEI) - (2 * gas * GWEI_TO_WEI)
+        self.logger.debug(f"Strata Balance lower bound: {lower_bound}")
+
+        assert balance >= lower_bound, "balance is not in the expected range"
+
+        # Mine blocks
+        btcrpc.proxy.generatetoaddress(12, UNSPENDABLE_ADDRESS)
+        time.sleep(1)
+
+        # Make sure that the balance in the BTC wallet is 10 BTC-ish
+        btc_balance = get_balance(withdraw_address, btc_url, btc_user, btc_password)
+        self.logger.debug(f"BTC balance: {btc_balance}")
+        btc_expected_balance = rollup_deposit_amount - 100_000_000
+        self.logger.debug(f"BTC expected balance lower bound: {btc_expected_balance}")
+        assert btc_balance >= btc_expected_balance, "BTC balance is not 10 BTC"
+
+        return True
+
+    def make_drt(self, ctx: flexitest.RunContext, el_address, musig_bridge_pk):
+        """
+        Deposit Request Transaction
+        """
+        # Get relevant data
+        btc = ctx.get_service("bitcoin")
+        seq = ctx.get_service("sequencer")
+        btcrpc: BitcoindClient = btc.create_rpc()
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+        seq_addr = seq.get_prop("address")
+
+        # Create the deposit request transaction
+        tx = bytes(
+            deposit_request_transaction(
+                el_address, musig_bridge_pk, btc_url, btc_user, btc_password
+            )
+        ).hex()
+        self.logger.debug(f"Deposit request tx: {tx}")
+
+        # Send the transaction to the Bitcoin network
+        txid = btcrpc.proxy.sendrawtransaction(tx)
+        self.logger.debug(f"sent deposit request with txid = {txid} for address {el_address}")
+        # this transaction is not in the bitcoind wallet, so we cannot use gettransaction
+        time.sleep(1)
+
+        # time to mature DRT
+        btcrpc.proxy.generatetoaddress(6, seq_addr)
+        time.sleep(3)
+
+        # time to mature DT
+        btcrpc.proxy.generatetoaddress(6, seq_addr)
+        time.sleep(3)
+
+    def make_withdraw(self, ctx: flexitest.RunContext, change_address_pk, gas=22_000):
+        """
+        Withdrawal Request Transaction in Strata's EVM.
+        """
+        reth = ctx.get_service("reth")
+        web3: Web3 = reth.create_web3()
+        web3.eth.default_account = web3.address
+        el_address = web3.address
+        self.logger.debug(f"EL address: {el_address}")
+        self.logger.debug(f"Bridge address: {PRECOMPILE_BRIDGEOUT_ADDRESS}")
+
+        data_bytes = bytes.fromhex(change_address_pk)
+
+        transaction = {
+            "from": el_address,
+            "to": PRECOMPILE_BRIDGEOUT_ADDRESS,
+            # FIXME: I don't know why the hell the math is wrong
+            #        and I need to manually add 5 zeros with 100_000
+            "value": DEFAULT_ROLLUP_PARAMS["deposit_amount"] * SATS_TO_WEI,
```

Comment:
Is this still applicable?

Reply:
No, will fix

Reply:
Done in 7a414eca5abf79b72ed921d945c06301005bca77